### PR TITLE
Implement collector image in curl

### PR DESCRIPTION
**Pull request by sweatybridge**

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
```
Comment (lawliet89) [inline start.sh:1]: You need to `chmod +x` this file.

File: start.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+set -euo pipefail
+
+result=$(curl -sL ${SRC_URL} | curl -sL -H "Authorization: Bearer ${API_KEY}" --data-binary @- ${DST_URL})
+
+if [ ! -z "$result" ]
```
Comment (lawliet89) [inline start.sh:6]: If you just run `curl -sL ${SRC_URL} | curl -sL -H "Authorization: Bearer ${API_KEY}" --data-binary @- ${DST_URL}` directly, the `set -euo pipefail` should fail the script in an error.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM alpine:3.8
+
+WORKDIR /app
+
+RUN apk add --no-cache curl
```
Comment (lawliet89) [inline Dockerfile:5]: Might want to add the [`ca-certificates`](https://pkgs.alpinelinux.org/package/v3.3/main/x86/ca-certificates) package too.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+COPY start.sh .
+
+ENV SRC_URL "http://ingress/metrics"
+ENV DST_URL "https://span/v1/metrics"
+ENV API_KEY "token"
+
+CMD ["./start.sh"]
```
Comment (lawliet89) [inline Dockerfile:13]: Might not need the `./`.

File: start.sh (hunk 2 above)
Comment (sweatybridge) [reply start.sh:6]: That only applies to the scenario when curl fails to connect. I'm also checking the http response by saving stdout of the post request. If post succeeds, result will be empty.

File: start.sh (hunk 1 above)
Comment (sweatybridge) [reply start.sh:1]: This file is stored in git with executable permission and COPY preserves the file permission so we don't necessarily need it.

File: start.sh (hunk 2 above)
Comment (sweatybridge) [reply start.sh:6]: If post returns 4xx, curl will exit with 0 nevertheless. But our api will return a message json.

File: Dockerfile (hunk 3 above)
Comment (sweatybridge) [reply Dockerfile]: You mean also add an env var for defining a path to self signed certs?

Comment (lawliet89) [reply Dockerfile:5]: No that is to install a list of "trusted" CAs. Like the usual suspects verisign, comodo, etc.

Comment (sweatybridge) [reply Dockerfile:5]: Yup, it comes as a dependency of curl.

**Pull request merged**